parastell/invessel_build.py

```diff
@@ -514,25 +515,38 @@ def _connect_ribs_with_tris_moab(self, rib1, rib2, reverse=False):
         return mb_tris
 
     def _generate_pymoab_verts(self):
-        """Generate MBVERTEX entities from rib loci in all surfaces"""
+        """Generate MBVERTEX entities from rib loci in all surfaces
+        (Internal function not intended to be called externally)
+        """
         [
             surface._generate_pymoab_verts(self.mbc)
             for surface in self.Surfaces.values()
         ]
 
-    def _generate_curved_surfaces_pydagmc(self):
+    def _generate_curved_surfaces_pydagmc(self, continuous_geom=False):
         """Generate the faceted representation of each curved surface and
         add it to the PyDAGMC model, remembering the surface ids. The sense
         of the triangles should point outward (increasing radial direction),
         with the exception of the first surface, which should point inward
         since the implicit complement is being used for the plasma chamber.
+        (Internal function not intended to be called externally)
+
+        Arguments:
+            continuous_geom (bool): flag indicating whether 360-degree,
+                continuous geometries should be generated.
         """
         self.curved_surface_ids = []
         surfaces = list(self.Surfaces.values())
         first_surface = surfaces[0]
         for surface in surfaces:
             mb_tris = []
-            for rib, next_rib in zip(surface.Ribs[0:-1], surface.Ribs[1:]):
+
+            if continuous_geom:
+                ribs = surface.Ribs[:-1] + [surface.Ribs[0]]
```

Comment:
This throws out the last rib - do we assume that it should be the same as rib 0 for a continuous geometry? Or do we assume that we don't care what it is?

Reply:
We assume it should be the same as rib 0